consent-ws/src/main/java/org/broadinstitute/consent/http/service/DatabaseConsentAPI.java

```diff
@@ -473,4 +473,17 @@ public Consent getByName(String name) throws UnknownIdentifierException {
         return consent;
     }
 
+    @Override
+    public Consent retrieveElectionDul(Integer electionId, Consent consent) {
+        Election election;
+        if (electionId != null) {
+            election = electionDAO.findElectionWithFinalVoteById(electionId);
+        } else {
+            election = electionDAO.getElectionWithFinalVoteByReferenceIdAndType(consent.consentId, ElectionType.TRANSLATE_DUL.getValue());
+        }
+        consent.setDataUseLetter(election.getDataUseLetter());
```

Comment:
Do you really want to be updating the consent? What if you just return the election object at this point?
My concern about updating the consent is what happens if the current election is canceled/archived and a new election is started? Do you remove the values that have been set for the DataUseLetter and name when this happens? (Meaning that now you have to keep these attributes in sync with the "correct" election).

Reply:
You are right Andrea, i didn't thought that. Please review again if now it looks better.